book/listings/gobject_signals/3/custom_button/imp.rs

```diff
@@ -23,27 +24,71 @@ impl ObjectSubclass for CustomButton {
 // ANCHOR: object_impl
 // Trait shared by all GObjects
 impl ObjectImpl for CustomButton {
-    fn constructed(&self, obj: &Self::Type) {
-        self.parent_constructed(obj);
-        obj.set_label(&self.number.borrow().to_string());
-    }
-
     fn signals() -> &'static [Signal] {
         static SIGNALS: Lazy<Vec<Signal>> = Lazy::new(|| {
             vec![Signal::builder(
                 // Signal name
                 "max-number-reached",
-                // Types of the values which will be sent to the receiver
+                // Types of the values which will be sent to the signal handler
                 &[i32::static_type().into()],
-                // Type of the value the receiver sends back
+                // Type of the value the signal handler sends back
                 <()>::static_type().into(),
             )
             .build()]
         });
         SIGNALS.as_ref()
     }
+    // ANCHOR_END: object_impl
+
+    fn properties() -> &'static [ParamSpec] {
+        static PROPERTIES: Lazy<Vec<ParamSpec>> = Lazy::new(|| {
+            vec![ParamSpec::new_int(
+                // Name
+                "number",
+                // Nickname
+                "number",
+                // Short description
+                "number",
+                // Minimum value
+                i32::MIN,
+                // Maximum value
+                i32::MAX,
+                // Default value
+                0,
+                // The property can be read and written to
+                ParamFlags::READWRITE,
+            )]
+        });
+        PROPERTIES.as_ref()
+    }
+
+    fn set_property(&self, _obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+        match pspec.name() {
+            "number" => {
+                let input_number = value.get().unwrap().unwrap();
+                self.number.replace(input_number);
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn get_property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
```

Comment:
this needs to be renamed per the latest changes

Reply:
Fixed